rdkit-stubs/Chem/BuildFragmentCatalog.pyi

```diff
@@ -1,98 +1,204 @@
 from _typeshed import Incomplete
+from collections.abc import Sequence
+from io import IOBase, TextIOWrapper
+import sys
+from typing import Optional
+import numpy as np
+import numpy.typing as npt
 from rdkit import RDConfig as RDConfig
 from rdkit.Chem import FragmentCatalog as FragmentCatalog
 from rdkit.Dbase.DbConnection import DbConnect as DbConnect
 from rdkit.ML import InfoTheory as InfoTheory
 
-def message(msg, dest=...) -> None: ...
+def message(msg, dest=sys.stdout) -> None: ...
 def BuildCatalog(
     suppl,
-    maxPts: int = ...,
-    groupFileName: Incomplete | None = ...,
-    minPath: int = ...,
-    maxPath: int = ...,
-    reportFreq: int = ...,
-): ...
+    maxPts: int = -1,
+    groupFileName: Optional[str] = None,
+    minPath: int = 2,
+    maxPath: int = 6,
+    reportFreq: int = 10,
+): 
+    """ 
+    builds a fragment catalog from a set of molecules in a delimited text block
+
+      **Arguments**
+
+        - suppl: a mol supplier
+
+        - maxPts: (optional) if provided, this will set an upper bound on the
+          number of points to be considered
+
+        - groupFileName: (optional) name of the file containing functional group
+          information
+
+        - minPath, maxPath: (optional) names of the minimum and maximum path lengths
+          to be considered
+
+        - reportFreq: (optional) how often to display status information
+
+      **Returns**
+
+        a FragmentCatalog
+
+    """
 def ScoreMolecules(
     suppl,
     catalog,
-    maxPts: int = ...,
-    actName: str = ...,
-    acts: Incomplete | None = ...,
-    nActs: int = ...,
-    reportFreq: int = ...,
-): ...
+    maxPts: int = -1,
+    actName: str = '',
+    acts: Optional[Sequence[int]] = None,
+    nActs: int = 2,
+    reportFreq: int = 10,
+) -> tuple[npt.NDArray[np.int32], list[list]]:
+    """ 
+    scores the compounds in a supplier using a catalog
+
+      **Arguments**
+
+        - suppl: a mol supplier
+
+        - catalog: the FragmentCatalog
+
+        - maxPts: (optional) the maximum number of molecules to be
+          considered
+
+        - actName: (optional) the name of the molecule's activity property.
+          If this is not provided, the molecule's last property will be used.
+
+        - acts: (optional) a sequence of activity values (integers).
+          If not provided, the activities will be read from the molecules.
+
+        - nActs: (optional) number of possible activity values
+
+        - reportFreq: (optional) how often to display status information
+
+      **Returns**
+
+        a 2-tuple:
+
+          1) the results table (a 3D array of ints nBits x 2 x nActs)
+
+          2) a list containing the on bit lists for each molecule
+
+    """
 def ScoreFromLists(
-    bitLists,
+    bitLists: Sequence,
```

Comment:
Should this be `Sequence[Sequence[bool]]`?

Reply:
Honestly speaking, I was lazy that I have not checked the detail of `bitLists`... Thank you for your polite suggestion. 

It seems `Sequence[Sequence[int]]` according to the unittests.

https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/UnitTestCatalog.py#L53-L55
```py
    self.list2Obls = [(0, 1, 2), (1, 3), (1, 4, 5), (1, 6, 7), (0, 8), (0, 6, 9), (0, 1, 2, 3, 10),
                      (0, 1, 2, 8, 11), (1, 3, 4, 5, 12), (1, 4, 5, 13), (1, 3, 6, 7, 14),
                      (0, 1, 6, 7, 9, 15)]
```

https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/UnitTestCatalog.py#L181-L188
```py
    scores, obls = BuildFragmentCatalog.ScoreMolecules(suppl, cat, acts=self.list2Acts,
                                                       reportFreq=20)
    for i in range(len(self.list2Obls)):
      assert tuple(
        obls[i]) == self.list2Obls[i], '%d: %s != %s' % (i, str(obls[i]), str(self.list2Obls[i]))

    scores2 = BuildFragmentCatalog.ScoreFromLists(obls, suppl, cat, acts=self.list2Acts,
                                                  reportFreq=20)
```